backend/src/main/java/com/google/sps/notifs/EmailNotifier.java

```diff
@@ -128,33 +77,93 @@ private static Message createMessageWithEmail(MimeMessage emailContent)
+    email.addRecipients(
+        RecipientType.TO,
+        new InternetAddress[] {
+          new InternetAddress(firstRecipientUsername + "@google.com"),
+          new InternetAddress(secondRecipientUsername + "@google.com")
+        });
+    email.setSubject("Ad-lib: Match Found");
+    email.setText(bodyText);
+    return email;
+  }
+
   /** Function that access its api and using it sends an email */
-  //   TODO(#35): Create a dummy email for ad lib itself to send emails.
   //   TODO(#36): Replace body to send real link to user instead of generic.
-  public void notifyUser() throws MessagingException, IOException, GeneralSecurityException {
-    NetHttpTransport HTTP_TRANSPORT = GoogleNetHttpTransport.newTrustedTransport();
+  public void sendMatchEmail(String firstMatchRecipientName, String secondMatchRecipientName)
+      throws MessagingException, IOException, GeneralSecurityException {
     MimeMessage email =
-        createEmail(
-            "Ad-Lib Meeting Found",
-            " Hey "
-                + recipientName
+        createEmailWithMultiRecipients(
+            firstMatchRecipientName,
+            secondMatchRecipientName,
+            "Congratulations"
                 + " Please Join your Ad-Lib meeting via the link below : \n"
                 + " http://meet.google.com/new");
     Message messageWithEmail = createMessageWithEmail(email);
     service.users().messages().send("me", messageWithEmail).execute();
   }
+
+  /** Function that access its api and using it sends an email */
+  //   TODO(#36): Replace body to send real link to user instead of generic.
+  public void sendExpiredEmail(String expiredRecipientName) throws MessagingException, IOException {
+    MimeMessage email =
+        createEmailWithSingleRecipient(
+            expiredRecipientName,
+            "Ad-lib: Meeting Query Expired",
+            " We apologize "
+                + expiredRecipientName
+                + "\n"
+                + " You're search for a match took longer than anticipated ! \n"
+                + " Please try again at a later time ! : \n");
+    Message messageWithEmail = createMessageWithEmail(email);
+    service.users().messages().send("me", messageWithEmail).execute();
+  }
+
+  /** Function that access its api and using it sends an email */
+  //   TODO(#36): Replace body to send real link to user instead of generic.
+  public void sendNoMatchEmail(String noMatchParticipantName)
+      throws MessagingException, IOException {
+    MimeMessage noMatchEmail =
+        createEmailWithSingleRecipient(
+            noMatchParticipantName,
+            "Ad-lib: No Match Found",
+            " Sorry "
+                + noMatchParticipantName
+                + " We could'nt find you a Match :( \n"
+                + " Please try again later.");
+    Message messageWithEmail = createMessageWithEmail(noMatchEmail);
+    service.users().messages().send("me", messageWithEmail).execute();
+  }
```

Comment:
The no match JSON response just tells the frontend that no match has been found YET so an email doesn't need to be sent here. Your correct that the expired JSON response needs to send an email though since this means we're taking that expired user out of the queue.

Reply:
Ahh, Okay